tests/unit/test_session.py

```diff
@@ -1113,10 +1174,25 @@ def unit_of_work(txn, *args, **kw):
                     metadata=[
                         ("google-cloud-resource-prefix", database.name),
                         ("x-goog-spanner-route-to-leader", "true"),
+                        (
+                            "x-goog-spanner-request-id",
+                            f"1.{REQ_RAND_PROCESS_ID}.{database.NTH_CLIENT.value}.1.2.1",
+                        ),
                     ],
-                )
-            ]
-            * 2,
+                ),
+                mock.call(
+                    session=self.SESSION_NAME,
+                    options=expected_options,
+                    metadata=[
+                        ("google-cloud-resource-prefix", database.name),
+                        ("x-goog-spanner-route-to-leader", "true"),
+                        (
+                            "x-goog-spanner-request-id",
+                            f"1.{REQ_RAND_PROCESS_ID}.{database.NTH_CLIENT.value}.1.2.2",
```

Comment:
Is `attempt` set to 2 here because the transaction is retried? If so, then that is not correct. The `attempt` number should only be increased if the RPC itself is being retried due to a retryable error for that (single) RPC. E.g. if a `ServiceUnavailable` error is returned.